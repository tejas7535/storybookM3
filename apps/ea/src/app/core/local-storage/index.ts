export * from './ea-parameters-local-storage-item.model';
export * from './local-storage.service';
