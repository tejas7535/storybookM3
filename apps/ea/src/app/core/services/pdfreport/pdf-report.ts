import jsPDF from 'jspdf'; // eslint-disable-line import/no-extraneous-dependencies

import { FontsLoaderService } from '../fonts-loader.service';
import { generateFooter } from './components/footer';
import { generateHeader } from './components/header';
import { VerticalLayout } from './components/vertical-layouter';
import {
  DocumentData,
  GeneratedDocument,
  ResultBlock,
  ResultReport,
} from './data';
export class PDFREport {
  constructor(
    private readonly docSettings: DocumentData,
    private readonly data: ResultReport,
    private readonly fontsLoaderService: FontsLoaderService
  ) {}

  async generate(): Promise<GeneratedDocument> {
    const doc = new jsPDF({ unit: 'pt' });
    this.fontsLoaderService.loadFonts(doc);
    const verticalLayout = new VerticalLayout(doc, {
      offsetTop: 85,
      offsetBottom: 90,
    }).add('methods', this.data.calculationMethods, {
      header: this.docSettings.calculationMethodsHeading,
    });

    const hasBearinxCalculation = this.hasCatalogCalculationResults();

    if (hasBearinxCalculation) {
      verticalLayout.add('inputs', this.data.calculationInput, {
        header: this.docSettings.inputSectionHeading,
      });
    }

    if (
      this.data.upstreamEmissions &&
      this.data.upstreamEmissions.data !== undefined
    ) {
      verticalLayout.add('upstream', this.data.upstreamEmissions, {
        props: { disclaimer: this.docSettings.co2disclaimer },
      });
    }

    if (this.data.ratingLife && this.data.ratingLife.data.length > 0) {
      verticalLayout.add('resultgrid', this.data.ratingLife);
    }

    if (
      this.data.overrollingFrequency &&
      this.data.overrollingFrequency.data.length > 0
    ) {
      verticalLayout.add('resultgrid', this.data.overrollingFrequency);
    }

    if (
      this.data.lubricationInfo &&
      this.data.lubricationInfo.data.length > 0
    ) {
      verticalLayout.add('resultgrid', this.data.lubricationInfo);
    }

    if (this.data.frictionalPowerloss) {
      verticalLayout.add('resultgrid', this.data.frictionalPowerloss);
    }

    if (hasBearinxCalculation) {
      verticalLayout.add('notices', this.data.notices);
    }

    verticalLayout.loop();

    this.generateHeaderAndFooterSectionsOnEveryPage(doc);

    return {
      designation: 'test',
      document: doc,
    };
  }
  private generateHeaderAndFooterSectionsOnEveryPage(doc: jsPDF) {
    const filteredPages = doc.internal.pages.filter(
      (page) => page !== undefined
    ); // for some reason undefined value is returned as a first page.

    const totalPages = filteredPages.length;
    for (let currentPage = 1; currentPage <= totalPages; currentPage += 1) {
      doc.setPage(currentPage);

      generateHeader(doc, this.docSettings);
      generateFooter(doc, currentPage, totalPages, this.docSettings);
    }
  }

  private hasCatalogCalculationResults(): ResultBlock<any> {
    const hasBearinxCalculation =
      this.data.ratingLife ||
      this.data.frictionalPowerloss ||
      this.data.lubricationInfo ||
      this.data.overrollingFrequency;

    return hasBearinxCalculation;
  }
}
