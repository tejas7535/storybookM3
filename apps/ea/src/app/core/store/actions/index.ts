export * from './calculation-parameters';
export * from './calculation-result';
export * from './product-selection';
export * from './settings';
export * from './storage-messages';
