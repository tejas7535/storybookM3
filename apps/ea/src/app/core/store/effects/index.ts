export * from './calculation-parameters';
export * from './calculation-result';
