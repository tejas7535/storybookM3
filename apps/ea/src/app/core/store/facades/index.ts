export * from './calculation-parameters/calculation-parameters.facade';
export * from './settings/settings.facade';
