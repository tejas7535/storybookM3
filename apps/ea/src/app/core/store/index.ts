export * from './facades';
export * from './reducers';
export * from './selectors';
export * from './store.module';
