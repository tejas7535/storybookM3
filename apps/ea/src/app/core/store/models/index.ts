export * from './calculation-parameter-groups.model';
export * from './calculation-parameters-state.model';
export * from './calculation-result-preview-data.model';
export * from './calculation-result-report-input.model';
export * from './calculation-result-state.model';
export * from './catalog-calculation-result-state.model';
export * from './co2-upstream-calculation-result-state.model';
export * from './friction-calculation-result-state.model';
export * from './product-selection-state.model';
export * from './settings-state.model';
