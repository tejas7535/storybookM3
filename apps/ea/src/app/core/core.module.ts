import { HTTP_INTERCEPTORS, HttpClientModule } from '@angular/common/http';
import { NgModule } from '@angular/core';
import {
  ErrorStateMatcher,
  ShowOnDirtyErrorStateMatcher,
} from '@angular/material/core';
import { MatIconRegistry } from '@angular/material/icon';
import { DomSanitizer } from '@angular/platform-browser';

import { environment } from '@ea/environments/environment';
import {
  AVAILABLE_LANGUAGES,
  FALLBACK_LANGUAGE,
  LANGUAGE_STORAGE_KEY,
} from '@ea/shared/constants/language';
import {
  TRANSLOCO_PERSIST_LANG_STORAGE,
  TranslocoPersistLangModule,
} from '@ngneat/transloco-persist-lang';

import { SharedTranslocoModule } from '@schaeffler/transloco';

import { HttpBearinxInterceptor } from './interceptor/http-bearinx.interceptor';
import { HttpCatalogWebApiInterceptor } from './interceptor/http-catalog-web-api.interceptor';
import { HttpCO2UpstreamInterceptor } from './interceptor/http-co2-upstream.interceptor';
import { StoreModule } from './store/store.module';

@NgModule({
  imports: [
    StoreModule,

    HttpClientModule,
    SharedTranslocoModule.forRoot(
      environment.production,
      AVAILABLE_LANGUAGES,
      undefined, // default -> undefined would lead to browser detection
      FALLBACK_LANGUAGE.id,
      LANGUAGE_STORAGE_KEY,
      true,
      !environment.localDev,
      undefined,
      `${environment.assetsPath}/i18n/`
    ),

    TranslocoPersistLangModule.forRoot({
      storageKey: LANGUAGE_STORAGE_KEY,
      storage: {
        provide: TRANSLOCO_PERSIST_LANG_STORAGE,
        useValue: localStorage,
      },
    }),
  ],

  providers: [
    {
      provide: HTTP_INTERCEPTORS,
      useClass: HttpBearinxInterceptor,
      multi: true,
    },
    {
      provide: HTTP_INTERCEPTORS,
      useClass: HttpCatalogWebApiInterceptor,
      multi: true,
    },
    {
      provide: HTTP_INTERCEPTORS,
      useClass: HttpCO2UpstreamInterceptor,
      multi: true,
    },
    { provide: ErrorStateMatcher, useClass: ShowOnDirtyErrorStateMatcher },
  ],

  exports: [StoreModule, SharedTranslocoModule],
})
export class CoreModule {
  public constructor(
    private readonly matIconRegistry: MatIconRegistry,
    private readonly sanitizer: DomSanitizer
  ) {
    this.registerEAIcons();
  }

  public registerEAIcons(): void {
    const iconSet: Record<string, string> = {
      co2: 'icon_CO2.svg',
      airwaves: 'icon_airwaves.svg',
      air: 'icon_air.svg',
      acute: 'icon_acute.svg',
      bolt: 'icon_bolt.svg',
      device_thermostat: 'icon_device_thermostat.svg',
      mop: 'icon_mop.svg',
      calculation: 'icon_calculations.svg',
      friction_load: 'icon_load_frictional_powerloss.svg',
      lubrication_parameters: 'icon_lubrication_parameters.svg',
      rating_life: 'icon_rpm_rating_life.svg',
      water_drop: 'icon_water_drop.svg',
    };
    for (const [name, url] of Object.entries(iconSet)) {
      const setUrl = this.sanitizer.bypassSecurityTrustResourceUrl(
        `${environment.assetsPath}/icons/${url}`
      );
      this.matIconRegistry.addSvgIcon(name, setUrl);
    }
  }
}
