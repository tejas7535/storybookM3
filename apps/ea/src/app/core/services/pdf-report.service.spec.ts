import { waitForAsync } from '@angular/core/testing';

import { of } from 'rxjs';

import { ResultReportLargeItem } from '@ea/calculation/calculation-result-report-large-items/result-report-large-item';
import { APP_STATE_MOCK } from '@ea/testing/mocks';
import { TranslocoService } from '@jsverse/transloco';
import { TranslocoLocaleService } from '@jsverse/transloco-locale';
import {
  createServiceFactory,
  mockProvider,
  SpectatorService,
} from '@ngneat/spectator/jest';
import { provideMockStore } from '@ngrx/store/testing';

import { ProductSelectionFacade } from '../store';
import { FontsLoaderService } from './fonts-loader.service';
import { PDFReportService } from './pdf-report.service';

const MOCK_RESULT_ITEMS: ResultReportLargeItem[] = [
  {
    value: 1,
    title: 'test 1',
    short: 'abc',
    unit: 'abc',
  },
  {
    value: 10,
    title: 'test 2',
    short: 'abc',
    unit: 'abc',
  },
];
const mockResultBlock = { header: 'Test', data: MOCK_RESULT_ITEMS };

describe('PDFReportService', () => {
  let pdfReportService: PDFReportService;
  let spectator: SpectatorService<PDFReportService>;

  const createService = createServiceFactory({
    service: PDFReportService,
    providers: [
      PDFReportService,
      provideMockStore({ initialState: APP_STATE_MOCK }),
      mockProvider(TranslocoService),
      mockProvider(FontsLoaderService),
      mockProvider(TranslocoLocaleService, {
        getLocale: jest.fn(() => 'en-US'),
        localizeDate: jest.fn(() => '1.1.1970'),
      }),
      mockProvider(ProductSelectionFacade, {
        bearingDesignation$: of('6226'),
      }),
    ],
  });

  beforeEach(() => {
    spectator = createService();
    pdfReportService = spectator.service;
  });

  it('should be created', () => {
    expect(pdfReportService).toBeDefined();
  });

  it('generate filename should generate a localized filename', waitForAsync(async () => {
    (
      pdfReportService['translocoService'].translate as jest.Mock
    ).mockReturnValue('Report');
    const filename = await pdfReportService.generateFilename();
    expect(pdfReportService['translocoService'].translate).toHaveBeenCalledWith(
      'pdfReport.filename',
      { designation: '6226', date: '1-1-1970' }
    );
    expect(filename).toEqual('Report');
  }));

  it('loadTranslationBlock should match snapshot', () => {
    expect(
      pdfReportService['loadTranslationBlock'](
        'ratingLife',
        mockResultBlock,
        'de'
      )
    ).toMatchSnapshot();
  });

  it('localize number formats should match snapshot', () => {
    expect(
      pdfReportService['localizeNumberFormats'](mockResultBlock)
    ).toMatchSnapshot();
  });

  it('translate emission should match snapshot', () => {
    expect(
      pdfReportService['translateEmission']('key', 'de')
    ).toMatchSnapshot();
  });
});
