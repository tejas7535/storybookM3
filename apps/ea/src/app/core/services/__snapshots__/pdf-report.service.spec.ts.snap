// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`PDFReportService loadTranslationBlock should match snapshot 1`] = `
Object {
  "data": Array [
    Object {
      "short": "abc",
      "title": undefined,
      "unit": "abc",
      "value": 1,
    },
    Object {
      "short": "abc",
      "title": undefined,
      "unit": "abc",
      "value": 10,
    },
  ],
  "header": "Test",
}
`;

exports[`PDFReportService localize number formats should match snapshot 1`] = `undefined`;

exports[`PDFReportService translate emission should match snapshot 1`] = `undefined`;
